actions/workflows/bwc_pkg_promote_all.yaml

```diff
@@ -1,176 +1,185 @@
-                - set_notify_success: <%
-                        $.promoted_bwc_enterprise and
-                        $.promoted_st2_auth_ldap and
-                        $.promoted_st2flow and
-                        $.promoted_bwc_ui %>
-
-        set_notify_failure:
-            action: std.noop
-            publish:
-                promoted: False
-                notify_color: "#FF0000"
-                notify_completion_channels: <% $.notify_failure_channels %>
-            on-success:
-                - notify_completion
-
-        set_notify_success:
-            action: std.noop
-            publish:
-                promoted: True
-                notify_color: "#008000"
-                notify_completion_channels: <% $.notify_channels %>
-            on-success:
-                - notify_completion
-
-        notify_completion:
-            with-items: channel in <% $.notify_completion_channels %>
-            action: slack.chat.postMessage
-            input:
-                channel: <% $.channel %>
-                text: ""
-                attachments: '[{"fallback": "[st2ci.bwc_pkg_promote_all: COMPLETED]", "title": "[st2ci.bwc_pkg_promote_all: COMPLETED]", "title_link": "<% $.webui_base_url %>/#/history/<% env().st2_execution_id %>/general", "text": "DISTRO: <% $.distro %>\nRELEASE: <% $.pkg_env %> <% $.release %>\nVERSION: <% coalesce($.version, ''latest'') %>\nPACKAGE PROMOTED:\n\tbwc-enterprise-packages=<% $.promoted_bwc_enterprise %>\n\tst2-auth-ldap=<% $.promoted_st2_auth_ldap %>\n\tst2flow=<% $.promoted_st2flow %>\n\tbwc-ui=<% $.promoted_bwc_ui %>", "color": "<% $.notify_color %>"}]'
-
-        notify_failure:
-            with-items: channel in <% $.notify_failure_channels %>
-            action: slack.chat.postMessage
-            input:
-                channel: <% $.channel %>
-                text: ""
-                attachments: '[{"fallback": "[st2ci.bwc_pkg_promote_all: FAILED]", "title": "[st2ci.bwc_pkg_promote_all: FAILED]", "title_link": "<% $.webui_base_url %>/#/history/<% env().st2_execution_id %>/general", "text": "DISTRO: <% $.distro %>\nRELEASE: <% $.pkg_env %> <% $.release %>\nVERSION: <% coalesce($.version, ''latest'') %>", "color": "#FF0000"}]'
+      channel: <% item(channel) %>
+      text: 'st2ci.bwc_pkg_promote_all: STARTED'
+      attachments: "[{\"fallback\": \"[st2ci.bwc_pkg_promote_all: STARTED]\", \"title\": \"[st2ci.bwc_pkg_promote_all: STARTED]\", \"title_link\": \"<% ctx().webui_base_url %>/#/history/<% ctx().st2.action_execution_id %>/general\", \"text\": \"DISTRO: <% ctx().distro %>\\nRELEASE: <% ctx().pkg_env %> <% ctx().release %>\\nVERSION: <% coalesce(ctx().version, 'latest') %>\", \"color\": \"#808080\"}]"
+    next:
+      - when: <% succeeded() %>
+        do:
+          - promote_all
+  promote_all:
+    next:
+      - do:
+          # These tasks do not properly join to process_completion when run in
+          # parallel, so we run them sequentially
```

Comment:
Responded to the issue at 
https://github.com/StackStorm/st2ci/issues/127#issuecomment-452570506. Looks like you should be able to fix and revert it back to parallel.

Reply:
Just to document what m4dcoder and I concluded:

This exposes a bug in Orquesta regarding merging contexts. For now, I have converted the workflow to promote packages sequentially, and I will circle back to this and refactor it to promote in parallel once the bug in Orquesta is resolved.

Issues:
* StackStorm/st2ci#127
* StackStorm/orquesta#112